docs/features/init_scripts.md

```diff
@@ -0,0 +1,51 @@
+# :material-script: Init scripts
+
+[Init scripts](https://docs.databricks.com/clusters/init-scripts.html) is a powerful tool in Databricks to setup the workflow environment before the workflow is running. A very common use case is to [setup the Python pip.conf](https://learn.microsoft.com/en-us/azure/databricks/kb/clusters/install-private-pypi-repo) if the workflow needs some private packages, then you don't need to declare it in each [pip install](https://docs.databricks.com/libraries/notebooks-python-libraries.html#install-a-private-package-with-credentials-managed-by-databricks-secrets-with-pip).
+
+## Appending custom init scripts
+
+In the dbx deployment file, if the key `new_cluster.policy_name` is specified, and the policy contains some init scripts. You can **append** additional init scripts by declaring them in the key `new_cluster.init_scripts`, and set an environment var `DBX_MERGE_INIT_SCRIPTS=true`. During `dbx deploy`, dbx will merge in order and with deduplication, the init scripts from the cluster policy and those from the key `new_cluster.init_scripts`.
+
+!!! tip
+    The `true` value for `DBX_MERGE_INIT_SCRIPTS` is case insensitive.
+
+One of the use cases is that as job cluster with [cluster pool](https://docs.databricks.com/clusters/instance-pools/index.html) is an isolated environment, two workflows sharing the the same pool won't be never distributed to the same node, we can define some shared init scripts in the cluster policy, such as `setup_pip.conf.sh`, etc., and we enable users to declare their own init scripts in the deployment file, for example `install_sql_driver.sh`. By this way, we keep the init scripts management easier instead of declaring all in the cluster policies.
+
+Here is a quick example how to use it:
+
+```bash
+export DBX_MERGE_INIT_SCRIPTS=true
+dbx deploy
+```
+
+And hereunder a simplified example of the [:material-file-code: deployment file](../reference/deployment.md):
+
+```json title="conf/deployment.json" linenums="1" hl_lines="10 12"
```

Comment:
Please use YAML format for examples in the documentation, they're way more readable.